### ISSUE TITLE
Offer strong table name length protection in schemas

### ISSUE THREAD

**Issue by jboreiko** [palantir/atlasdb#2023]
Currently I can write schemas that will render my product inoperable against Cassandra / Postgres backed KVS without notifying me (as I can be testing only against InMem / H2).  I would prefer that these limits are enforced at schema generation time and I'm allowed to flag in my schema generation that I'm okay accepting that I will not be able to function with certain backends.

You can think of this as a stop gap prior to implementing table mapping for Cassandra / Postgres.  

The error currently manifests as an unclear error when running ETE tests against Cassandra.

```
! com.palantir.atlasdb.keyvalue.impl.TableMappingNotFoundException: Unable to resolve full name for table reference <greater than 48 character table name>
! at com.palantir.atlasdb.keyvalue.impl.AbstractTableMappingService.getMappedTableName(AbstractTableMappingService.java:64)
! at com.palantir.atlasdb.keyvalue.impl.TableRemappingKeyValueService.getRows(TableRemappingKeyValueService.java:277)
! ... 47 common frames omitted
! Causing: java.lang.IllegalArgumentException: com.palantir.atlasdb.keyvalue.impl.TableMappingNotFoundException: Unable to resolve full name for table reference <greater than 48 character table name>
! at com.palantir.atlasdb.keyvalue.impl.TableRemappingKeyValueService.getRows(TableRemappingKeyValueService.java:282)
! at com.palantir.atlasdb.keyvalue.impl.NamespaceMappingKeyValueService.getRows(NamespaceMappingKeyValueService.java:100)
! at com.palantir.atlasdb.keyvalue.impl.ProfilingKeyValueService.lambda$getRows$31(ProfilingKeyValueService.java:344)
! at com.palantir.atlasdb.keyvalue.impl.ProfilingKeyValueService.maybeLog(ProfilingKeyValueService.java:205)
! at com.palantir.atlasdb.keyvalue.impl.ProfilingKeyValueService.getRows(ProfilingKeyValueService.java:344)
! at com.palantir.atlasdb.keyvalue.impl.ForwardingKeyValueService.getRows(ForwardingKeyValueService.java:101)
! at com.palantir.atlasdb.keyvalue.impl.TracingKeyValueService.getRows(TracingKeyValueService.java:278)
! at sun.reflect.GeneratedMethodAccessor72.invoke(Unknown Source)
! at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
! at java.lang.reflect.Method.invoke(Method.java:498)
! at com.palantir.tritium.proxy.InvocationEventProxy.execute(InvocationEventProxy.java:138)
! at com.palantir.tritium.proxy.InvocationEventProxy.instrumentInvocation(InvocationEventProxy.java:115)
! at com.palantir.tritium.proxy.InvocationEventProxy.handleInvocation(InvocationEventProxy.java:95)
! at com.google.common.reflect.AbstractInvocationHandler.invoke(AbstractInvocationHandler.java:84)
! at com.sun.proxy.$Proxy61.getRows(Unknown Source)
! at com.palantir.atlasdb.keyvalue.impl.ValidatingQueryRewritingKeyValueService.getRows(ValidatingQueryRewritingKeyValueService.java:172)
! at com.palantir.atlasdb.transaction.impl.SnapshotTransaction.getRows(SnapshotTransaction.java:344)
! at com.palantir.atlasdb.transaction.impl.SerializableTransaction.getRows(SerializableTransaction.java:138)
! at com.palantir.atlasdb.transaction.impl.ForwardingTransaction.getRows(ForwardingTransaction.java:53)
```

From this it is not clear to the develop what the problem is.  A more descriptive error would also greatly improve the situation.

**Comment by hsaraogi** [comment]
This should be a part of removing support for JDBC KVS to enforce length names for migration. This can be done alongside unifying table name mapping.

**Comment by amarzoca** [comment]
Fixed in #2220
